### PR TITLE
Add tests for Guard functions

### PR DESCRIPTION
**Pull request by toddmowen**
As a first step towards fixing #285, these tests characterise the current behaviour of the utility functions in `au.com.cba.omnia.maestro.core.hdfs.Guard`.

### CODE REVIEW

File: maestro-core/src/test/scala/au/com/cba/omnia/maestro/core/hdfs/GuardSpec.scala [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+        s"file:$dir/user/b1"
+        // excludes "b2" because it's not a directory
+      )
+    }
+  }
+
+  def expandPaths_skipsProcessed = {
+    withEnvironment(path(getClass.getResource("/hdfs-guard").toString)) {
+      Guard.expandPaths(s"$dir/user/c*") must_== List(
+        s"file:$dir/user/c",
```
Comment (SamRoberts) [inline maestro-core/src/test/scala/au/com/cba/omnia/maestro/core/hdfs/GuardSpec.scala]: This is testing not just the contents of the list, but also the order in which `Guard.expandPaths` returns the contents. You might be interested in checking out the matchers available in specs 2 here: https://code.google.com/p/specs/wiki/MatchersGuide. In particular, the matchers for iterables, which demonstrates `list1 must haveTheSameElementsAs(list2)`.

Comment (SamRoberts) [inline maestro-core/src/test/scala/au/com/cba/omnia/maestro/core/hdfs/GuardSpec.scala]: (And likewise in other places in the file)

**Comment by SamRoberts** [comment]
Really good to see these tests :smile:

File: maestro-core/src/test/scala/au/com/cba/omnia/maestro/core/hdfs/GuardSpec.scala [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+  }
+
+  def createFlagFile_createsPROCESSED = {
+    withEnvironment(path(getClass.getResource("/hdfs-guard").toString)) {
+      Guard.createFlagFile(List(s"$dir/user/a1", s"$dir/user/b1"))
+      new File(s"$dir/user/a1/_PROCESSED").exists() must beTrue
+      new File(s"$dir/user/b1/_PROCESSED").exists() must beTrue
+    }
+  }
+
```
Comment (stephanh) [inline maestro-core/src/test/scala/au/com/cba/omnia/maestro/core/hdfs/GuardSpec.scala]: Superfluous line.

**Comment by stephanh** [comment]
@toddmowen very nice PR. I don't think we have had such a clean first PR before!

:+1: once Sam's comment and the failing test are addressed. Let me know if you want some help squashing the commits.

File: maestro-core/src/test/scala/au/com/cba/omnia/maestro/core/hdfs/GuardSpec.scala (hunk 1 above)
Comment (toddmowen) [inline maestro-core/src/test/scala/au/com/cba/omnia/maestro/core/hdfs/GuardSpec.scala]: Thanks Sam, this should resolve the failed test in Travis CI.

**Pull request merged**